lib/puppet/catalog-diff/compilecatalog.rb

```diff
@@ -49,19 +49,21 @@ def get_catalog_from_puppetdb(node_name, server)
       Puppet.debug("Getting PuppetDB catalog for #{node_name}")
       require 'puppet/util/puppetdb'
       server_url = Puppet::Util::Puppetdb.config.server_urls[0]
-      port = server_url.port
-      use_ssl = port != 8080
-      connection = Puppet::Network::HttpPool.http_instance(server_url.host, port, use_ssl)
```

Comment:
Previously we checked the port. PuppetDB uses 8080 for HTTP and 8081 for HTTPS. But the config always returns a full URI: https://puppet.com/docs/puppetdb/7/puppetdb_connection.html#server-urls
Instead of hardcoding the Port/protocol here I think we should use what the config file returns.